src/main/java/com/szeloch/jan/parkingsystemmanagement/rest/controller/parkingclient/ParkingClientsController.java

```diff
@@ -0,0 +1,192 @@
+import com.szeloch.jan.parkingsystemmanagement.rest.model.Vehicle;
+import com.szeloch.jan.parkingsystemmanagement.rest.repository.ParkingClientsRepository;
+import com.szeloch.jan.parkingsystemmanagement.rest.repository.ParkingTicketsRepository;
+import com.szeloch.jan.parkingsystemmanagement.rest.repository.VehiclesRepository;
+import com.szeloch.jan.parkingsystemmanagement.utils.ParkingTicketsUtil;
+import org.springframework.hateoas.Resource;
+import org.springframework.hateoas.ResourceSupport;
+import org.springframework.hateoas.Resources;
+import org.springframework.hateoas.VndErrors;
+import org.springframework.hateoas.mvc.ControllerLinkBuilder;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.*;
+
+import java.net.URI;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping(value = "/parking-clients", produces = "application/hal+json")
+public class ParkingClientsController {
+
+    private final ParkingClientsRepository clientsRepository;
+
+    private final VehiclesRepository vehiclesRepository;
+
+    private final ParkingTicketsRepository ticketsRepository;
+
+    private final ParkingTicketsUtil parkingTicketsUtil;
+
+    public ParkingClientsController(final ParkingClientsRepository clientsRepository, final VehiclesRepository vehiclesRepository, final ParkingTicketsRepository ticketsRepository, final ParkingTicketsUtil parkingTicketsUtil) {
+        this.clientsRepository = clientsRepository;
+        this.vehiclesRepository = vehiclesRepository;
+        this.ticketsRepository = ticketsRepository;
+        this.parkingTicketsUtil = parkingTicketsUtil;
+    }
+
+    @GetMapping
+    public Resources<ParkingClientResource> findAll() {
+
+        final List<ParkingClientResource> parkingClients = clientsRepository
+                .findAll()
+                .stream()
+                .map(ParkingClientResource::new)
+                .collect(Collectors.toList());
+
+        return new Resources<>(parkingClients,
+                linkTo(methodOn(ParkingClientsController.class).findAll()).withSelfRel());
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<ParkingClientResource> findOne(@PathVariable final long id) {
+
+        return clientsRepository
+                .findById(id)
+                .map(parkingClient -> ResponseEntity.ok(new ParkingClientResource(parkingClient)))
+                .orElseThrow(() -> new ParkingClientNotFoundException(id));
+    }
+
+    @PostMapping
+    public ResponseEntity<ParkingClientResource> newParkingClient(@RequestBody final ParkingClient newParkingClient) {
+
+        ParkingClientResource resource = new ParkingClientResource(clientsRepository.save(newParkingClient));
```

Comment:
... Ale zapis do bazy w kontrolerze to już kiepsko.

Reply:
Mogłem utworzyć klasy Service i w nich przeprowadzać operacje na bazie z wykorzystaniem wstrzykniętych repository, ale klasy te nie robiły by nic więcej, uznałem, że w takim małym testowym projekcie mogę przeprowadzić te operacje bezpośrednio w controllerze.

Reply:
No właśnie, bo brakuje jakichś walidacji na przykład :) Jakichś reguł biznesowych. Wtedy serwisy by były potrzebne.